Update libmesh: VTK 9.4 configure fixes and nlohmann patch

loganharbour opened the pull request:
refs #30349

VTK 9.4 is not discoverable by libmesh configure. This update patches that.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2829204/), step Versioner verify on 86440c9 wanted to post the following:

### Versioner templates

Found 16 templates, 0 failed


### Versioner versions

Found 10 packages, 3 changed, 0 failed

| package     | status   | hash    | to hash   | version            | to version         |
|-------------|----------|---------|-----------|--------------------|--------------------|
| libmesh-vtk | CHANGE   | b994539 | 887cc4d   | 9.4.2 build 0      | 9.4.2 build 1      |
| libmesh     | CHANGE   | cbc75b9 | a69a11b   | 2025.04.17 build 0 | 2025.04.17 build 1 |
| moose-dev   | CHANGE   | 1b20943 | 1488f40   | 2025.04.17         | 2025.04.21         |

milljm reviewed (approved):
👍

loganharbour committed:
Update libmesh

Contains configure fixes for VTK 9.4

refs #30349

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit b0ea29004a60fa0ce8a351c57f5f722f737ab297
+Subproject commit 8d1b4c43d1ab2283d4702aa86cf040d2ae2ec5c0
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

loganharbour committed:
Add VTK patch for nlohmann issue refs #30372

loganharbour committed:
Bump vtk, libmesh, and moose-dev

loganharbour committed:
Update versioner_hashes

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2830377/), step Versioner verify on 34db190 wanted to post the following:

### Versioner templates

Found 16 templates, 0 failed


### Versioner versions

Found 10 packages, 3 changed, 0 failed

| package     | status   | hash    | to hash   | version            | to version         |
|-------------|----------|---------|-----------|--------------------|--------------------|
| libmesh-vtk | CHANGE   | b994539 | 59e681e   | 9.4.2 build 0      | 9.4.2 build 1      |
| libmesh     | CHANGE   | cbc75b9 | e0eb0da   | 2025.04.17 build 0 | 2025.04.22 build 0 |
| moose-dev   | CHANGE   | 1b20943 | 8a3a8dd   | 2025.04.17         | 2025.04.22         |

roystgnr committed:
Attempted workaround for VTK 9.4 bug

This isn't necessary with 9.3 and older, and we're hoping they're fixing
the problem with 9.5, but this is what it takes to get us to compile
against a non-patched 9.4.

loganharbour committed:
Ignore VTKFile version for iga tests

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2830621/), step Versioner verify on fe3ffa4 wanted to post the following:

### Versioner templates

Found 16 templates, 0 failed


### Versioner versions

Found 10 packages, 3 changed, 0 failed

| package     | status   | hash    | to hash   | version            | to version         |
|-------------|----------|---------|-----------|--------------------|--------------------|
| libmesh-vtk | CHANGE   | b994539 | 59e681e   | 9.4.2 build 0      | 9.4.2 build 1      |
| libmesh     | CHANGE   | cbc75b9 | e0eb0da   | 2025.04.17 build 0 | 2025.04.22 build 0 |
| moose-dev   | CHANGE   | 1b20943 | 8a3a8dd   | 2025.04.17         | 2025.04.22         |

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2830672/), step Versioner verify on f47a89c wanted to post the following:

### Versioner templates

Found 16 templates, 0 failed


### Versioner versions

Found 10 packages, 3 changed, 0 failed

| package     | status   | hash    | to hash   | version            | to version         |
|-------------|----------|---------|-----------|--------------------|--------------------|
| libmesh-vtk | CHANGE   | b994539 | 59e681e   | 9.4.2 build 0      | 9.4.2 build 1      |
| libmesh     | CHANGE   | cbc75b9 | e0eb0da   | 2025.04.17 build 0 | 2025.04.22 build 0 |
| moose-dev   | CHANGE   | 1b20943 | 8a3a8dd   | 2025.04.17         | 2025.04.22         |

loganharbour commented:
Merging despite CI completion due to priorities; same tests will run on next.

Have verified that the moose-dev environment passes framework+module tests for:

- conda moose-dev with patched vtk
- apptainer moose-dev with patched vtk
- apptainer moose-dev without patched vtk
